src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -87,33 +75,26 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
                 asyncValueFactory: () => MergeVersionsAsync(identity, completed));
         }
 
-        public async Task<IPackageSearchMetadata> GetLatestPackageMetadataAsync(PackageIdentity identity,
-            bool includePrerelease, CancellationToken cancellationToken)
+        public async Task<IPackageSearchMetadata> GetLatestPackageMetadataAsync(
+            PackageIdentity identity,
+            NuGetProject project,
+            bool includePrerelease, 
+            CancellationToken cancellationToken)
         {
-            VersionRange allowedVersions = VersionRange.All;
+            var allowedVersions = VersionRange.All;
 
-            // at this time, allowed version range should be applied at project level. 
-            // For solution level, we will apply allowed version range for all the selected projects in detail control model
-            if (!_isSolution)
-            {
-                if (_packageReferences == null)
-                {
-                    // get all package references for all the projects and cache locally
-                    var packageReferencesTasks = _projects.Select(project => project.GetInstalledPackagesAsync(cancellationToken));
-                    _packageReferences = (await Task.WhenAll(packageReferencesTasks)).SelectMany(p => p).Where(p => p != null);
-                }
-
-                // filter package references for current package identity
-                var matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.PackageIdentity.Id, identity.Id));
+            // get all package references for all the projects and cache locally
+            var packageReferences = await project.GetInstalledPackagesAsync(cancellationToken);
 
-                // Allowed version range for current package across all selected projects
-                var allowedVersionsRange = matchedPackageReferences.Select(r => r.AllowedVersions).Where(v => v != null).ToArray();
+            // filter package references for current package identity
+            var matchedPackageReferences = packageReferences
+                .Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.PackageIdentity.Id, identity.Id));
 
-                if (allowedVersionsRange.Length > 0)
-                {                    
-                    allowedVersions = allowedVersionsRange[0];
-                }
-            }
+            // Allowed version range for current package across all selected projects
+            // Picks the first non-default range
+            allowedVersions = matchedPackageReferences
+                .Select(r => r.AllowedVersions)
+                .FirstOrDefault(v => v != null) ?? VersionRange.All;
 
```

Comment:
When a package is installed in multiple projects, and out of those multiple projects only one project has defined allowedVersion, then it will restrict showing latest version of that package at solution level even though there're projects which could install that latest version.